org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/NexusScanFileManager.java

```diff
@@ -329,9 +330,13 @@ private void addLegacyPerScanMonitors(ScanModel model, Collection<String> scanna
 		}
 		
 		try {
-			scannable.setMonitorRole(MonitorRole.PER_SCAN);
+			if (scannable.getMonitorRoleDefault() != MonitorRole.PER_SCAN) {
+				logger.warn("Not setting {} DefaultMonitorRole to {}, currently {}", scannable.getName(), MonitorRole.PER_SCAN, scannable.getMonitorRoleDefault());
+			} else if (false) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)